### PR TITLE
Implement unregistering renderers

### LINKED ISSUE #56: API for unregistering inputs/outputs/transformations

### PR DESCRIPTION
**Pull request by wkozyra95**
Closes #56

### CODE REVIEW

File: compositor_render/src/registry.rs [hunk 1]
```diff
@@ -11,6 +11,14 @@ pub enum RegisterError {
     },
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("Failed to unregister a {item_type}. The \"{renderer_id}\" {item_type} does not exist.")]
+pub struct UnregisterError {
+    item_type: &'static str,
+    renderer_id: RendererId,
+}
```
Comment (WojciechBarczynski) [inline compositor_render/src/registry.rs:19]:
```suggestion
pub struct UnregisterItemNotFoundError {
    item_type: &'static str,
    renderer_id: RendererId,
}
```

There are a lot of other unregister errors

File: compositor_render/src/registry.rs [hunk 2]
```diff
@@ -56,4 +64,14 @@ impl<T: Clone> RendererRegistry<T> {
...
     }
+
+    pub(crate) fn unregister(&mut self, id: &RendererId) -> Result<(), UnregisterError> {
+        match self.registry.remove(id) {
+            Some(_) => Ok(()),
+            None => Err(UnregisterError {
+                item_type: self.registry_type.registry_item_name(),
+                renderer_id: id.clone(),
+            }),
+        }
```
Comment (WojciechBarczynski) [inline compositor_render/src/registry.rs:75]: Use `.ok_or`

**Review by WojciechBarczynski — APPROVED** [review]

File: compositor_render/src/registry.rs (hunk 1 above)
Comment (wkozyra95) [reply compositor_render/src/registry.rs:19]: I would prefer to leave it like this for consistency with RegisterError.

File: compositor_render/src/registry.rs (hunk 2 above)
Comment (wkozyra95) [reply compositor_render/src/registry.rs:75]: ok_or maps None to Err, but here I need both map None to Err and Some(value) to `Ok(())`

**Pull request merged**